src/AppInstallerCLICore/Commands/ResumeCommand.cpp

```diff
@@ -0,0 +1,135 @@
+                        break;
+                    }
+                }
+
+                if (!commandFound)
+                {
+                    THROW_HR_MSG(E_UNEXPECTED, "Command to resume not found.");
+                }
+            }
+
+            return std::move(commandToResume);
+        }
+    }
+
+    using namespace std::string_view_literals;
+    using namespace Execution;
+
+    std::vector<Argument> ResumeCommand::GetArguments() const
+    {
+        return {
+            Argument::ForType(Execution::Args::Type::ResumeId),
+        };
+    }
+
+    Resource::LocString ResumeCommand::ShortDescription() const
+    {
+        return { Resource::String::ResumeCommandShortDescription };
+    }
+
+    Resource::LocString ResumeCommand::LongDescription() const
+    {
+        return { Resource::String::ResumeCommandLongDescription };
+    }
+
+    Utility::LocIndView ResumeCommand::HelpLink() const
+    {
+        return "https://aka.ms/winget-command-resume"_liv;
+    }
+
+    void ResumeCommand::ExecuteInternal(Execution::Context& context) const
+    {
+        const auto& resumeId = context.Args.GetArg(Execution::Args::Type::ResumeId);
+
+        if (!std::filesystem::exists(Checkpoints::CheckpointManager::GetCheckpointDatabasePath(resumeId)))
+        {
+            context.Reporter.Error() << Resource::String::ResumeIdNotFoundError(Utility::LocIndView{ resumeId }) << std::endl;
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_RESUME_ID_NOT_FOUND);
+        }
+
+        Execution::Context resumeContext = context.CreateEmptyContext();
+        std::optional<Checkpoint<AutomaticCheckpointData>> foundAutomaticCheckpoint = resumeContext.LoadCheckpoint(std::string{ resumeId });
+        if (!foundAutomaticCheckpoint.has_value())
+        {
+            context.Reporter.Error() << Resource::String::ResumeStateDataNotFoundError << std::endl;
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_INVALID_RESUME_STATE);
+        }
+
+        Checkpoint<AutomaticCheckpointData> automaticCheckpoint = foundAutomaticCheckpoint.value();
+
+        const auto& checkpointClientVersion = automaticCheckpoint.Get(AutomaticCheckpointData::ClientVersion, {});
+        if (checkpointClientVersion != AppInstaller::Runtime::GetClientVersion().get())
+        {
+            context.Reporter.Error() << Resource::String::ClientVersionMismatchError(Utility::LocIndView{ checkpointClientVersion }) << std::endl;
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_CLIENT_VERSION_MISMATCH);
+        }
+
+        const auto& checkpointCommand = automaticCheckpoint.Get(AutomaticCheckpointData::Command, {});
+        AICLI_LOG(CLI, Info, << "Resuming command: " << checkpointCommand);
+        std::unique_ptr<Command> commandToResume = FindCommandToResume(checkpointCommand);
+
+        for (const auto& fieldName : automaticCheckpoint.GetFieldNames(AutomaticCheckpointData::Arguments))
```

Comment:
For future maintainability, it would be better if this code that reads the values out were next to the code that writes the values in.  Even if that means a standalone helper, I would much rather have them in the same file (and even have the functions next to each other).